app/src/main/shared/components/InternalLink.tsx

```diff
@@ -8,8 +8,15 @@ interface Props {
 }
 
 export class InternalLink extends React.Component<Props, undefined> {
+    static contextTypes = {
+        router: React.PropTypes.object
+    }
     render() {
         const url = appSettings.publicPath + this.props.href;
-        return <Link href={ url } onClick={ this.props.onClick }>{ this.props.children }</Link>
+        if (this.context.router) {
+            return <Link to={url} onClick={this.props.onClick}>{this.props.children}</Link>
+        } else {
+            return <a href={url} onClick={this.props.onClick}>{this.props.children}</a>
+        }
```

Comment:
When would an `InternalLink` have or not have a contextual router? Maybe a comment here explaining that?

Reply:
Why is this so different from `ButtonLink`?